src/gpu/d3d12/SDL_gpu_d3d12.c

```diff
@@ -5299,18 +5287,25 @@ static void D3D12_INTERNAL_AllocateCommandBuffer(
         return;
     }
 
-    D3D12CommandBuffer **resizedAvailableCommandBuffers = SDL_realloc(
-        renderer->availableCommandBuffers,
-        sizeof(D3D12CommandBuffer *) * (renderer->availableCommandBufferCapacity + 1));
+    if (renderer->availableCommandBufferCount >= renderer->availableCommandBufferCapacity) {
```

Comment:
Should we use the macro here too? I noticed that the old pointer doesn't get updated for a few lines so that may actually be a problem for async access if realloc chooses a new address.